packages/remark/types/test.ts

```diff
@@ -0,0 +1,13 @@
+import remark = require('remark')
+
+interface PluginOptions {
+  example: boolean
+}
+
+const plugin = (options?: PluginOptions) => {}
+
+remark().use(plugin)
+remark().use(plugin, {example: true})
+remark().use({settings: {commonmark: true}})
```

Comment:
Looking this over, I wonder if Unified itself could have a generic so the settings can be validated. :thought_balloon: 


@wooorm is it accurate that presets has three forms?

```ts
// Three ways presets should be done
// plugins added via preset
unified().use({ plugins: [] })
// plugins and settings
unified().use({ plugins: [], settings: {} })
// parser and stringifier with settings
remark().use({ settings: {} })

//======
// This shouldn't happen
unified().use({ settings: {} })
```

Reply:
Opened https://github.com/unifiedjs/unified/pull/59 as one potential way to allow `remark` to declare settings typings for the parser and stringifier bundle.

Reply:
```
unified().use({ settings: {} })
unified().use({})
```

...should not happen if remark-parse / remark-stringify (or some other plugin that uses `settings`) is not used. If a plugin defines that it uses settings somehow, these two cases would be OK (they both reset the settings).